lgtm warnings

kilobyte opened the pull request:
These are all minor issues.

There's one bug left: passing cache size as an int, despite it being possibly as big as 8GB.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/pmdk/5281)
<!-- Reviewable:end -->

pbalcer commented:
windows failed because of ctime change...
https://stackoverflow.com/questions/35750940/ctime-r-on-msvc

kilobyte commented:
I've dropped the ctime commit, to avoid complicating the examples.  On the other hand, the same issue with localtime touches only our code — thus I've added necessary #ifdefs there.

An interesting quirk is that C11 has localtime_s but Microsoft's version is incompatible with that (reversed order of args).  localtime_r is in C23 which MSVC doesn't support yet.

wlemkows commented:
@kilobyte do you want this changes only for master?

kilobyte commented:
I'm not sure.  The fixes are mostly to satisfy warnings, and to reduce confusion (variable name clashes).

The only real fixes here are:
* making localtime() thread/delayed use -safe
* stopping an example from crashing if modfied

If you'd prefer to target one of the stable versions, could you please tell me which?

wlemkows reviewed (commented):
Changes:

•	making localtime() thread/delayed use -safe
•	stopping an example from crashing if modfied

 for 1.8+

> *[Reviewable](https://reviewable.io/reviews/pmem/pmdk/5281#-:-MguNFP594mzEV_IWfcc:b-i7acjk)* status: 0 of 14 files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

kilobyte commented:
Ok, retargetted.  I had to cherry-pick a commit from master to let stable-1.8 even compile with modern toolchain — but stable not building is something we really want to fix.

wlemkows reviewed (commented):
Today we talked with the team about this fix and we decided to apply the changes for release 1.10+.
Could you make such a change?  @kilobyte really sorry for the inconvenience.

> *[Reviewable](https://reviewable.io/reviews/pmem/pmdk/5281#-:-Mgz5ArcDJGKu4iLYv02:bu2epoy)* status: 0 of 17 files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

kilobyte commented:
✓
